x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -595,4 +597,28 @@ static DeprecationIssue checkClusterRoutingAllocationIncludeRelocationsSetting(f
             DeprecationIssue.Level.CRITICAL
         );
     }
+
+    static DeprecationIssue checkFrozenCacheLeniency(final Settings settings,
+                                                     final PluginsAndModules pluginsAndModules,
+                                                     final ClusterState clusterState,
+                                                     final XPackLicenseState licenseState) {
+        final String cacheSizeSettingKey = "xpack.searchable.snapshot.shared_cache.size";
+        Setting<ByteSizeValue> cacheSizeSetting =  Setting.byteSizeSetting(cacheSizeSettingKey,  ByteSizeValue.ZERO);
+        if (cacheSizeSetting.exists(settings)) {
+            ByteSizeValue cacheSize = cacheSizeSetting.get(settings);
+            if (cacheSize.getBytes() > 0) {
```

Comment:
Just double checking here because I know that -1 usually means "disabled": Is this expected to be 0 or negative, or should it be always negative?

Reply:
On 8.0 it looks like we are not throwing an error if it is 0 -- https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/FrozenCacheService.java#L128